internal/cmd/organizations/list.go

```diff
@@ -35,36 +34,12 @@ func listOrgsCommand() *cobra.Command {
 				return fmt.Errorf("failed to list organizations: %w", err)
 			}
 
-			// TODO: We should look at abstracting the formatting here into a library
-			//       that can be used by multiple commands needing to offer multiple
-			//       output formats from a command.
-			switch outputFormat {
-			case "yaml":
-				marshaller := &protoyaml.MarshalOptions{
-					Indent: 2,
-				}
-				output, err := marshaller.Marshal(listOrgs)
-				if err != nil {
-					return fmt.Errorf("failed to list organizations: %w", err)
-				}
-				fmt.Print(string(output))
-			case "json":
-				output, err := protojson.Marshal(listOrgs)
-				if err != nil {
-					return fmt.Errorf("failed to list organizations: %w", err)
-				}
-				fmt.Print(string(output))
-			case "table":
-				orgTable := table.New("DISPLAY NAME", "RESOURCE ID")
-				if len(listOrgs.Organizations) == 0 {
-					fmt.Printf("No organizations found")
-				} else {
-					for _, org := range listOrgs.Organizations {
-						orgTable.AddRow(org.DisplayName, org.OrganizationId)
-					}
-				}
-				orgTable.Print()
+			headers := []any{"DISPLAY NAME", "RESOURCE ID"}
+			var rowData [][]any
+			for _, org := range listOrgs.Organizations {
+				rowData = append(rowData, []any{org.DisplayName, org.OrganizationId})
 			}
+			output.CLIPrint(os.Stdout, outputFormat, listOrgs, headers, rowData)
```

Comment:
@sullivtr what are your thoughts on passing in a "table formatter" function (potentially leveraging generics?) that could be used to control which fields are outputted in the table? 

Something like this (untested):

```go
output.CLIPrint(os.Stdout, outputFormat, listOrgs, func() ([]string, output.RowFormatter)  {
  return []string{"DISPLAY NAME", "RESOURCE ID"}, func(row *resourcemanagerpb.Organization) []string {
    return []string{org.DisplayName, org.OrganizationId}
  }
})
```

Reply:
Looking into this now. I do think using generics might not work as nicely as you expect without a common interface between all of of the list types we might be operating on tho (since you have to pass `T int64 | float64` style piped T args).

Reply:
For example, I got the UX paired down to this, which isnt much simpler from a UX perspective, but it does create some guardrails around the table format objectives. 

```go
output.CLIPrintGeneric(os.Stdout, outputFormat, listOrgs, func() (output.ColumnFormatter, output.RowFormatterFunc) {
    return output.ColumnFormatter{"DISPLAY NAME", "RESOURCE ID"}, func() output.RowFormatter {
        var rowData output.RowFormatter
	for _, org := range listOrgs.Organizations {
	    rowData = append(rowData, []any{org.DisplayName, org.OrganizationId})
	}
	return rowData
  }
})
```

Reply:
I'd be happy with the approach you took here. I think it helps address some of the UX concerns I had.

Reply:
Perfect. Just pushed that change up, along with an update to tests. I also added an err check on the CLIPrint check that I was previously ignoring.